File: Week 03/id_701/LeetCode_200_701.py
class Solution:
    def numIslands(self, grid: list) -> int:
        def dfs(grid: list, i: int, j: int):
            if i < 0 or j < 0 or i >= len(grid) or j >= len(grid[0]) or grid[i][j] != '1':
                return
            grid[i][j] = '#'

            dfs(grid, i-1, j)
            dfs(grid, i+1, j)
            dfs(grid, i, j-1)
            dfs(grid, i, j+1)

        count = 0
        if grid:
            for i in range(len(grid)):
                for j in range(len(grid[i])):
                    if grid[i][j] == '1':
                        dfs(grid, i, j)
                        count += 1
        return count

File: Week 06/id_701/leetcode_200_701.cs
public class Solution
{
    public int NumIslands(char[][] grid)
    {
        int count = 0;
        if (grid != null)
        {
            for (int i = 0; i < grid.Length; i++)
            {
                for (int j = 0; j < grid[i].Length; j++)
                {
                    if (grid[i][j] == '1')
                    {
                        DFS(grid, i, j);
                        count += 1;
                    }
                }
            }
        }
        return count;
    }

    private void DFS(char[][] grid, int x, int y)
    {
        if (x < 0 || x > grid.Length-1 || y < 0 || y > grid[0].Length-1 || grid[x][y] != '1')
        {
            return;
        }
        grid[x][y] = '#';

        DFS(grid, x + 1, y);
        DFS(grid, x - 1, y);
        DFS(grid, x, y + 1);
        DFS(grid, x, y - 1);
    }
}